src/pollen.py

```diff
@@ -88,6 +86,14 @@ def check_rng_file(self, charm_state):
             charm_state: Pollen charm's CharmState instance.
         """
         file = Path("/etc/default/rng-tools-debian")
+        charm_state.rng_tools_file = (
+            None
+            if not file.exists()
+            or 2
+            > file.read_text(encoding="utf-8").count(
+                'RNGDOPTIONS="--fill-watermark=90% --feed-interval=1"'
+            )
+            else 'RNGDOPTIONS="--fill-watermark=90% --feed-interval=1"'
+        )
```

Comment:
I wouldn't do it that way, but if others agree with this implem I'm fine.
@jdkandersson Can I get your input ?

I would rather have a property (initialized in the constructor) or a method, that gets the file content within the the CharmState
And probably have the logic to write in the file based on that state property/method content

Reply:
Yes I agree with Gregory, this state shouldn't be assigned like this, it should get calculated by the constructor or within a function as suggested by Gregory